pkg/scheduler/util/utils_test.go

```diff
@@ -173,3 +174,72 @@ func TestRemoveNominatedNodeName(t *testing.T) {
 		})
 	}
 }
+
+func TestPatchPodStatus(t *testing.T) {
+	tests := []struct {
+		name           string
+		podName        string
+		secrets        []v1.LocalObjectReference
+		statusToUpdate v1.PodStatus
+	}{
+		{
+			name:    "Should update pod conditions successfully",
+			podName: "pod2",
+			secrets: []v1.LocalObjectReference{{Name: "foo"}},
+			statusToUpdate: v1.PodStatus{
+				Conditions: []v1.PodCondition{
+					{
+						Type:   v1.PodScheduled,
+						Status: v1.ConditionFalse,
+					},
+				},
+			},
+		},
+		{
+			name:    "Should update pod conditions successfully on a pod Spec with secrets with empty name",
```

Comment:
Add a comment with a link to the relevant k/k issue, so that future readers understand why this test case is important.